zerver/lib/actions.py

```diff
@@ -1181,6 +1181,12 @@ def do_deactivate_stream(
     # Get the affected user ids *before* we deactivate everybody.
     affected_user_ids = can_access_stream_user_ids(stream)
 
+    Subscription.objects.filter(
+        recipient__type=Recipient.STREAM,
+        recipient__type_id=stream.id,
+        role=Subscription.ROLE_STREAM_ADMINISTRATOR,
+    ).update(role=Subscription.ROLE_MEMBER)
```

Comment:
Hmm, ideally we'd want this to still record something in RealmAuditLog about the change in administrator status, e.g. by calling the `do_...` function for changing role, rather than a raw database query.

We haven't settled plans for reactivating deactivated streams, but if/when we do, we may want a record of who the administrators had been.